File: src/test/java/com/mikea/bayes/belief/PruneClusterGraphTest.java
package com.mikea.bayes.belief;

import com.mikea.bayes.ProbabilitySpace;
import com.mikea.bayes.Var;
import com.mikea.bayes.VarSet;
import org.gga.graph.impl.DataGraphImpl;
import org.junit.Test;

import static com.mikea.bayes.VarSet.newVarSet;
import static org.junit.Assert.assertEquals;

/**
 * @author mike.aizatsky@gmail.com
 */
public class PruneClusterGraphTest {
    @Test
    public void testDoesntKillGraph() throws Exception {
        ProbabilitySpace space = new ProbabilitySpace();

        Var a = space.newVar("a", 2);
        Var b = space.newVar("b", 2);
        Var c = space.newVar("c", 2);

        VarSet abSet = newVarSet(a, b);
        VarSet acSet = newVarSet(a, c);
        VarSet aSet = newVarSet(a);

        DataGraphImpl.Builder<VarSet, VarSet> builder = new DataGraphImpl.Builder<VarSet, VarSet>(VarSet.class, VarSet.class, false);
        builder
                .addNode(abSet)
                .addNode(acSet)
                .addNode(aSet)
                .addEdge(aSet, abSet, aSet)
                .addEdge(aSet, acSet, aSet);

        ClusterGraph prunedGraph = PruneClusterGraph.prune(new ClusterGraphImpl(space, builder.build()));
        assertEquals(
                "DataGraphImpl{isDirected=false, [{a, b}, {a, c}], [\n" +
                        "    {a, b}<->{a, c}:{a}\n" +
                        "]}",
                prunedGraph.toString()
        );
    }
}

File: src/test/java/com/mikea/bayes/belief/PruneClusterGraphTest.scala
package com.mikea.bayes.belief

import org.junit.Test

/**
 * @author mike.aizatsky@gmail.com
 */
class PruneClusterGraphTest {
  @Test def testDoesntKillGraph {
    ???
/*
    val space: ProbabilitySpace = new ProbabilitySpace
    val a: Var = space.newVar("a", 2)
    val b: Var = space.newVar("b", 2)
    val c: Var = space.newVar("c", 2)
    val abSet: VarSet = newVarSet(a, b)
    val acSet: VarSet = newVarSet(a, c)
    val aSet: VarSet = newVarSet(a)
    val builder: Nothing = new Nothing(classOf[VarSet], classOf[VarSet], false)
    builder.addNode(abSet).addNode(acSet).addNode(aSet).addEdge(aSet, abSet, aSet).addEdge(aSet, acSet, aSet)
    val prunedGraph: ClusterGraph = PruneClusterGraph.prune(new ClusterGraphImpl(space, builder.build))
    assertEquals("DataGraphImpl{isDirected=false, [{a, b}, {a, c}], [\n" + "    {a, b}<->{a, c}:{a}\n" + "]}", prunedGraph.toString)
*/
  }
}